compiler/rustc_lint/src/types.rs

```diff
@@ -727,7 +728,48 @@ declare_lint! {
     "proper use of libc types in foreign item definitions"
 }
 
-declare_lint_pass!(ImproperCTypesDefinitions => [IMPROPER_CTYPES_DEFINITIONS]);
+declare_lint! {
+    /// In it's platform C ABI, the AIX target follows the power alignment rule
+    /// for structs. This rule applies the natural alignment of the first struct
+    /// field if it is a floating-point type that is greater than 4-bytes, or
+    /// is an aggregate whose recursively "first" field is a floating-point
+    /// type greater than 4-bytes.
+    /// The alignment for subsequent fields of the associated structs use an
+    /// alignment value where the floating-point type is aligned on a 4-byte
+    /// boundary.
+    /// In this case, "natural alignment" (detailed in
+    /// https://www.ibm.com/docs/en/xl-c-and-cpp-aix/16.1?topic=data-using-alignment-modes#alignment)
+    /// refers to the floating-point type being aligned to a multiple of its
+    /// size (for example, the size of a double is 8-bytes, and so is 8-byte
+    /// aligned).
+    ///
+    /// The power alignment rule for structs need for C compatibility is
+    /// currently unimplemented within repr(C) in the compiler, so a warning is
+    /// produced in these situations.
```

Comment:
Functionally, these types wrap every float in `#[repr(packed(4))]`, and we would need to figure out some way to handle that.
```suggestion
    /// The power alignment rule for structs needed for C compatibility is
    /// unimplementable within `repr(C)` in the compiler without building in
    /// handling of references to packed fields and infectious nested layouts,
    /// so a warning is produced in these situations.
```